core/templates/components/forms/custom-forms-directives/object-editor.directive.ts

```diff
@@ -18,136 +18,258 @@
+import { TranslatableSetOfNormalizedStringEditorComponent } from 'objects/templates/translatable-set-of-normalized-string-editor.component';
+import { TranslatableSetOfUnicodeStringEditorComponent } from 'objects/templates/translatable-set-of-unicode-string-editor.component';
+import { UnicodeStringEditorComponent } from 'objects/templates/unicode-string-editor.component';
+import { IntEditorComponent } from 'objects/templates/int-editor.component';
+import { LoggerService } from 'services/contextual/logger.service';
+import { ComponentRef } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { downgradeComponent } from '@angular/upgrade/static';
+const EDITORS = {
+  'algebraic-expression': AlgebraicExpressionEditorComponent,
+  'boolean': BooleanEditorComponent,
+  'code-string': CodeStringEditorComponent,
+  'coord-two-dim': CoordTwoDimEditorComponent,
+  'custom-osk-letters': CustomOskLettersEditorComponent,
+  'drag-and-drop-positive-int': DragAndDropPositiveIntEditorComponent,
+  filepath: FilepathEditorComponent,
+  fraction: FractionEditorComponent,
+  graph: GraphEditorComponent,
+  html: HtmlEditorComponent,
+  'image-with-regions': ImageWithRegionsEditorComponent,
+  'int': IntEditorComponent,
+  'list-of-sets-of-translatable-html-content-ids':
+    ListOfSetsOfTranslatableHtmlContentIdsEditorComponent,
+  'list-of-tabs': ListOfTabsEditorComponent,
+  'list-of-unicode-string': ListOfUnicodeStringEditorComponent,
+  'math-equation': MathEquationEditorComponent,
+  'math-expression-content': MathExpressionContentEditorComponent,
+  'set-of-unicode-string': SetOfUnicodeStringEditorComponent,
+  'music-phrase': MusicPhraseEditorComponent,
+  'number-with-units': NumberWithUnitsEditorComponent,
+  'nonnegative-int': NonnegativeIntEditorComponent,
+  'normalized-string': NormalizedStringEditorComponent,
+  'numeric-expression': NumericExpressionEditorComponent,
+  'parameter-name': ParameterNameEditorComponent,
+  'position-of-terms': PositionOfTermsEditorComponent,
+  'positive-int': PositiveIntEditorComponent,
+  'ratio-expression': RatioExpressionEditorComponent,
+  real: RealEditorComponent,
+  'sanitized-url': SanitizedUrlEditorComponent,
+  'set-of-algebraic-identifier': SetOfAlgebraicIdentifierEditorComponent,
+  'set-of-translatable-html-content-ids':
+    SetOfTranslatableHtmlContentIdsEditorComponent,
+  'skill-selector': SkillSelectorEditorComponent,
+  'subtitled-html': SubtitledHtmlEditorComponent,
+  'subtitled-unicode': SubtitledUnicodeEditorComponent,
+  svg: SvgEditorComponent,
+  'translatable-html-content-id': TranslatableHtmlContentIdEditorComponent,
+  'translatable-set-of-normalized-string':
+    TranslatableSetOfNormalizedStringEditorComponent,
+  'translatable-set-of-unicode-string':
+    TranslatableSetOfUnicodeStringEditorComponent,
+  'unicode-string': UnicodeStringEditorComponent,
+};
+
+interface ObjectEditor {
+  alwaysEditable: string;
+  initArgs: unknown;
```

Comment:
Object is a very generic type. Something like `initArgs.a` would be allowed but not when the type is unknown. Unknown is type safe any. We need to have runtime checks to use something like `initArgs.a`. Since I didn't have much time to determine the type of this variable and I didn't want to set the type as any because it will lead to further development on this file to sort of skip some type checks. Hence I opted for the very safe unknown type that would error on any wrong use of the variable and would enforce us to use the right types for any further modifications.